src/UNIV2LPOracle.t.sol

```diff
@@ -546,6 +549,33 @@ contract UNIV2LPOracleTest is DSTest {
         assertTrue(daiEthLPOracle.pass());                           // Fail pass
     }
 
+    // This test examines the gas costs of a common pattern which needs
+    // to be maximally gas-optimized.
+    function test_gas_pass_poke() public {
+        uint256 preGas;
+        uint256 diffGas;
+
+        require(daiEthLPOracle.pass());
+
+        preGas = gasleft();
+        if (daiEthLPOracle.pass()) {
+            daiEthLPOracle.poke();
+        }
+        diffGas = preGas - gasleft();
+        assertTrue(diffGas <= 88408);  // Will need to be updated for the Berlin hardfork.
+        log_named_uint("pass+poke gas", diffGas);
+
+        require(!daiEthLPOracle.pass());
+
+        preGas = gasleft();
+        if (daiEthLPOracle.pass()) {
+            daiEthLPOracle.poke();
+        }
+        diffGas = preGas - gasleft();
+        assertTrue(diffGas <= 3465);  // Will need to be updated for the Berlin hardfork.
+        log_named_uint("pass-only gas", diffGas);
```

Comment:
![Screenshot from 2021-03-23 09-23-24](https://user-images.githubusercontent.com/16065447/112161802-77cbfd00-8bb9-11eb-89b9-53aa86a5eaff.png)

Reply:
See reply to the similar comment.